test/TritonGPU/amd/amd-canonicalize-pointers.mlir

```diff
@@ -1315,3 +1315,63 @@ module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 4 : i32, ttg.targ
 // CHECK:           tt.store %[[VAL_26]], %[[VAL_24]], %[[VAL_29]] : tensor<64x!tt.ptr<i8>, #[[$ATTR_4]]>
 // CHECK:           tt.return
 // CHECK:         }
+
+// -----
+
+module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 4 : i32, "ttg.threads-per-warp" = 32 : i32} {
+  tt.func public @asin_kernel(%arg0: !tt.ptr<i64>) attributes {noinline = false} {
+    %c1024_i32 = arith.constant 1024 : i32
+    %0 = tt.get_program_id x : i32
+    %1 = arith.muli %0, %c1024_i32 : i32
+    %2 = tt.make_range {end = 1024 : i32, start = 0 : i32} : tensor<1024xi32>
+    %3 = tt.splat %1 : i32 -> tensor<1024xi32>
+    %4 = arith.addi %3, %2 : tensor<1024xi32>
+    %5 = tt.splat %arg0 : !tt.ptr<i64> -> tensor<1024x!tt.ptr<i64>>
+    %6 = tt.addptr %5, %4 : tensor<1024x!tt.ptr<i64>>, tensor<1024xi32>
+    %7 = tt.extern_elementwise %6 {libname = "", libpath = "", pure = false, symbol = "foo"} : (tensor<1024x!tt.ptr<i64>>) -> tensor<1024xi64>
+    tt.return
+  }
+}
+
+// CHECK-LABEL:   tt.func public @asin_kernel(
+// CHECK-SAME:                                %[[VAL_0:.*]]: !tt.ptr<i64>) attributes {noinline = false} {
+// CHECK:           %[[VAL_1:.*]] = arith.constant 1024 : i32
+// CHECK:           %[[VAL_2:.*]] = tt.get_program_id x : i32
+// CHECK:           %[[VAL_3:.*]] = arith.muli %[[VAL_2]], %[[VAL_1]] : i32
+// CHECK:           %[[VAL_4:.*]] = tt.make_range {end = 1024 : i32, start = 0 : i32} : tensor<1024xi32>
+// CHECK:           %[[VAL_5:.*]] = tt.addptr %[[VAL_0]], %[[VAL_3]] : !tt.ptr<i64>, i32
+// CHECK:           %[[VAL_6:.*]] = tt.splat %[[VAL_5]] : !tt.ptr<i64> -> tensor<1024x!tt.ptr<i64>>
+// CHECK:           %[[VAL_7:.*]] = tt.addptr %[[VAL_6]], %[[VAL_4]] : tensor<1024x!tt.ptr<i64>>, tensor<1024xi32>
+// CHECK:           %[[VAL_8:.*]] = tt.extern_elementwise %[[VAL_7]] {libname = "", libpath = "", pure = false, symbol = "foo"} : (tensor<1024x!tt.ptr<i64>>) -> tensor<1024xi64>
+// CHECK:           tt.return
+// CHECK:         }
+
+// -----
+
+module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 4 : i32, "ttg.threads-per-warp" = 32 : i32} {
+  tt.func public @inline_asm(%arg0: !tt.ptr<i8>, %arg1: !tt.ptr<i8>) attributes {noinline = false} {
+    %0 = tt.make_range {end = 512 : i32, start = 0 : i32} : tensor<512xi32>
+    %1 = tt.splat %arg0 : !tt.ptr<i8> -> tensor<512x!tt.ptr<i8>>
+    %2 = tt.addptr %1, %0 : tensor<512x!tt.ptr<i8>>, tensor<512xi32>
+    %3 = tt.load %2 : tensor<512x!tt.ptr<i8>>
+    %4 = tt.elementwise_inline_asm "shl.b32 $0, $0, 3;" {constraints = "=r,r", packed_element = 4 : i32, pure = true} %3 : tensor<512xi8> -> tensor<512xi8>
```

Comment:
Note, this is PTX which is "weird" since this is and AMD pass but we don't have any AMD examples so I just grabbed the [first one I found](https://github.com/triton-lang/triton/blob/84f09062015374119f5a7c4f5912c2f551b79408/test/Conversion/tritongpu_to_llvm.mlir#L1809).